### PR TITLE
gengateway: allow opting out patch feature

### PR DESCRIPTION
**Pull request by glerchundi**
PTAL @johanbrandhorst

### CODE REVIEW

File: protoc-gen-grpc-gateway/gengateway/template_test.go [hunk 1]
```diff
@@ -383,7 +383,7 @@ func TestApplyTemplateRequestWithClientStreaming(t *testing.T) {
 				},
 			},
 		}
-		got, err := applyTemplate(param{File: crossLinkFixture(&file), RegisterFuncSuffix: "Handler"}, descriptor.NewRegistry())
+		got, err := applyTemplate(param{File: crossLinkFixture(&file), RegisterFuncSuffix: "Handler", AllowPatchFeature: true}, descriptor.NewRegistry())
```
Comment (johanbrandhorst) [inline protoc-gen-grpc-gateway/gengateway/template_test.go]: Can we add a test case for the new case as well?

File: protoc-gen-grpc-gateway/main.go [hunk 2]
```diff
@@ -32,6 +32,7 @@ var (
 	pathType                   = flag.String("paths", "", "specifies how the paths of generated files are structured")
 	allowRepeatedFieldsInBody  = flag.Bool("allow_repeated_fields_in_body", false, "allows to use repeated field in `body` and `response_body` field of `google.api.http` annotation option")
 	repeatedPathParamSeparator = flag.String("repeated_path_param_separator", "csv", "configures how repeated fields should be split. Allowed values are `csv`, `pipes`, `ssv` and `tsv`.")
+	allowPatchFeature          = flag.Bool("allow_patch_feature", true, "determines whether to use PATCH feature involving update masks (FieldMask.")
```
[the remark this answers is not in the document]
Comment (glerchundi) [reply protoc-gen-grpc-gateway/main.go]: Good catch.

File: protoc-gen-grpc-gateway/gengateway/template_test.go (hunk 1 above)
Comment (glerchundi) [reply protoc-gen-grpc-gateway/gengateway/template_test.go]: Sure.

Comment (glerchundi) [reply protoc-gen-grpc-gateway/gengateway/template_test.go]: Where do you prefer me to add the test? The test infra seems a little bit weak so does it make if i double-check this inside `TestApplyTemplateRequestWithoutClientStreaming`?

Comment (johanbrandhorst) [reply protoc-gen-grpc-gateway/gengateway/template_test.go]: Could we add a new test case which singles out this case? I'm on mobile ATM so don't have s great view of the files.

Comment (glerchundi) [reply protoc-gen-grpc-gateway/gengateway/template_test.go]: Done.

File: protoc-gen-grpc-gateway/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (glerchundi) [reply protoc-gen-grpc-gateway/main.go]: Done.

**Comment by glerchundi** [comment]
PTAL @johanbrandhorst, it seems like a bazel test is not passing but i think it's not something related to this change.

**Comment by glerchundi** [comment]
Thanks Andrew for taking a look. Let me know if something is needed from my
side.

Once merged will appreciate if you can cut a release as this is blocking us
from upgrading (as we thought a minor upgrade would not contain a backward
incompatible change).

**Comment by achew22** [comment]
https://github.com/grpc-ecosystem/grpc-gateway/pull/841 should fix that. If you could please rebase it would be hugely appreciated. Thanks!

**Comment by glerchundi** [comment]
@achew22 rebased, lets see :)

Thanks!

**Pull request merged**